mysql-test/suite/rpl/r/rpl_show_slave_hosts.result

```diff
@@ -17,4 +17,198 @@ connection master;
+Rpl_semi_sync_master_status	ON
+show status like 'Rpl_semi_sync_master_no_tx';
+Variable_name	Value
+Rpl_semi_sync_master_no_tx	0
+show status like 'Rpl_semi_sync_master_yes_tx';
+Variable_name	Value
+Rpl_semi_sync_master_yes_tx	0
+show status like 'Rpl_semi_sync_master_clients';
+Variable_name	Value
+Rpl_semi_sync_master_clients	1
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Semisync	File	Position	Gtid
+2	localhost	SLAVE_PORT	1	OFF	master-bin.000001	674	0-1-2
+# Check with new replication server with id 3
+connection slave2;
+RESET SLAVE;
+CHANGE MASTER TO master_host='127.0.0.1',master_port=MASTER_PORT,master_user='root';
+START SLAVE IO_THREAD;
+include/wait_for_slave_io_to_start.inc
+connection master;
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Semisync	File	Position	Gtid
+3	slave2	SLAVE_PORT2	1	OFF		0	
+2	localhost	SLAVE_PORT	1	OFF	master-bin.000001	674	0-1-2
+# Enable semisync replication on server with id 3
+connection slave2;
+stop slave;
+reset slave;
+set global rpl_semi_sync_slave_enabled = 1;
+start slave;
+show variables like 'rpl_semi_sync_slave_enabled';
+Variable_name	Value
+rpl_semi_sync_slave_enabled	ON
+include/wait_for_slave_io_to_start.inc
+connection master;
+show status like 'Rpl_semi_sync_master_clients';
+Variable_name	Value
+Rpl_semi_sync_master_clients	2
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Semisync	File	Position	Gtid
+3	slave2	SLAVE_PORT2	1	OFF		0	
+2	localhost	SLAVE_PORT	1	OFF	master-bin.000001	674	0-1-2
+create table t1(a int);
+INSERT INTO t1 SET a=1;
+show status like 'Rpl_semi_sync_master_status';
+Variable_name	Value
+Rpl_semi_sync_master_status	ON
+show status like 'Rpl_semi_sync_master_no_tx';
+Variable_name	Value
+Rpl_semi_sync_master_no_tx	0
+show status like 'Rpl_semi_sync_master_yes_tx';
+Variable_name	Value
+Rpl_semi_sync_master_yes_tx	0
+show status like 'Rpl_semi_sync_master_clients';
+Variable_name	Value
+Rpl_semi_sync_master_clients	2
+show status like 'Rpl_semi_sync_master_clients';
+Variable_name	Value
+Rpl_semi_sync_master_clients	2
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Semisync	File	Position	Gtid
+3	slave2	SLAVE_PORT2	1	OFF	master-bin.000001	1018	0-1-4
+2	localhost	SLAVE_PORT	1	OFF	master-bin.000001	1018	0-1-4
+# Replica 1 changed rpl_semi_sync_enabled value to 0 (asynchronous replication)
+connection slave2;
+stop slave;
+reset slave;
+set global rpl_semi_sync_slave_enabled = 0;
+start slave;
+[ slave status should be OFF ]
+show status like 'Rpl_semi_sync_slave_status';
+Variable_name	Value
+Rpl_semi_sync_slave_status	OFF
+connection master;
+show status like 'Rpl_semi_sync_master_clients';
+Variable_name	Value
+Rpl_semi_sync_master_clients	1
+[ master status should be ON ]
+show status like 'Rpl_semi_sync_master_status';
+Variable_name	Value
+Rpl_semi_sync_master_status	ON
+SHOW REPLICA HOSTS;
+Server_id	Host	Port	Master_id	Semisync	File	Position	Gtid
+3	slave2	SLAVE_PORT2	1	OFF		0	
+3	slave2	SLAVE_PORT2	1	OFF	master-bin.000001	1018	0-1-4
```

Comment:
This is interesting. Perhaps a bug (although not related to the task at-hand). @andrelkin if a slave re-connects while a master is awaiting a semi-sync ACK, `show slave hosts` will report the same slave multiple times, until the timeout finishes and the thread finishes up. Should we add logic (in a different ticket) on slave connect, to check for any existing connections with that server, and clean them up if they exist?